### PR TITLE
[cli] Make filesystem globbing faster in `vercel dev`

### PR DESCRIPTION
**Pull request by TooTallNate**
The current code uses `getAllProjectFiles()` which globs for every file
in the project, not considering the `.vercelignore` file and default
ignore list. For example, every file in the `node_modules` directory is
selected, just for `vercel dev` to manually ignore them afterwards,
which is very slow and wasteful. For a simple Next.js project, this
globbing was taking over 3 seconds on my machine 🤯!

Solution is to use the `staticFiles()` function which is the same as
`vercel dev` uses at boot-up time. This function properly considers the
ignore list and thus is much faster, and the manual filtering is no
longer necessary by `vercel dev`. For the same simple Next.js project,
this function takes on average 10 milliseconds.

Also did a bit of cleanup and removed the `getAllProjectFiles()`
function since it is no longer used anywhere.

### CODE REVIEW

File: packages/now-cli/src/util/dev/server.ts [hunk 1]
```diff
@@ -532,15 +536,6 @@ export default class DevServer {
       this.readJsonFile<NowConfig>(configPath),
     ]);
 
-    const allFiles = await getAllProjectFiles(this.cwd, this.output);
```
Comment (TooTallNate) [inline packages/now-cli/src/util/dev/server.ts:535]: This is the main fix.

File: packages/now-cli/src/util/dev/server.ts [hunk 2]
```diff
@@ -555,6 +550,11 @@ export default class DevServer {
       const featHandleMiss = true; // enable for zero config
       const { projectSettings, cleanUrls, trailingSlash } = config;
 
+      const opts = { output: this.output, isBuilds: true };
```
Comment (TooTallNate) [inline packages/now-cli/src/util/dev/server.ts:554]: This is also the main fix.

File: packages/now-cli/src/util/get-files.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
-import { resolve, join } from 'path';
```
Comment (TooTallNate) [inline packages/now-cli/src/util/get-files.ts:1]: Except for the removal of the `getAllProjectFiles()` function, the rest of the changes in this file are only cosmetic.

**Review by AndyBitz — APPROVED** [review]
🏅

**Review by AndyBitz — APPROVED** [review]

File: packages/now-cli/src/util/get-files.ts [hunk 4]
```diff
@@ -415,24 +411,6 @@ export async function docker(
...
- * @return {Array} of {String}s with the found files
- */
-export async function getAllProjectFiles(cwd: string, { debug }: Output) {
-  // We need a slash at the end to remove it later on from the matched files
-  const current = join(resolve(cwd), '/');
-  debug(`Searching files inside of ${current}`);
-
-  const list = await glob('**', { cwd: current, absolute: true, nodir: true });
-
-  // We need to replace \ with / for windows
```
Comment (styfle) [inline packages/now-cli/src/util/get-files.ts:432]: Does the new function work with Windows? I assume so but it’s good to test it out first to confirm.

**Review by styfle — APPROVED** [review]

File: packages/now-cli/src/util/get-files.ts (hunk 4 above)
Comment (TooTallNate) [reply packages/now-cli/src/util/get-files.ts:432]: It does. The `relative()` helper function that is used in `dev/server.ts` does this same replacement.

**Review by pacocoursey — APPROVED** [review]

**Review by MaxLeiter — APPROVED** [review]

**Pull request merged**